test_vm/tests/commit_post_test.rs

```diff
@@ -0,0 +1,426 @@
+                            to: *STORAGE_POWER_ACTOR_ADDR,
+                            method: PowerMethod::UpdatePledgeTotal as u64,
+                            ..Default::default()
+                        }]),
+                        ..Default::default()
+                    },
+                    ExpectInvocation {
+                        to: *REWARD_ACTOR_ADDR,
+                        method: RewardMethod::UpdateNetworkKPI as u64,
+                        ..Default::default()
+                    },
+                ]),
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: *STORAGE_MARKET_ACTOR_ADDR,
+                method: MarketMethod::CronTick as u64,
+                ..Default::default()
+            },
+        ]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+    // pcd is released ip is added
+    let balances = v.get_miner_balance(id_addr);
+    assert!(balances.initial_pledge > TokenAmount::zero());
+    assert!(balances.pre_commit_deposit == TokenAmount::zero());
+
+    // power unproven so network stats are the same
+    let p_st = v.get_state::<PowerState>(*STORAGE_POWER_ACTOR_ADDR).unwrap();
+    assert_eq!(TokenAmount::zero(), p_st.total_bytes_committed);
+    assert!(p_st.total_pledge_collateral > TokenAmount::zero());
+    let (dline_info, p_idx, v) = advance_to_proving_deadline(v, id_addr, sector_number);
+
+    // submit post
+    let st = v.get_state::<MinerState>(id_addr).unwrap();
+    let sector = st.get_sector(v.store, sector_number).unwrap().unwrap();
+    let partitions = vec![PoStPartition {
+        index: p_idx,
+        skipped: fvm_ipld_bitfield::UnvalidatedBitField::Validated(BitField::new()),
+    }];
+    let sector_power = power_for_sector(seal_proof.sector_size().unwrap(), &sector);
+    submit_windowed_post(&v, worker, id_addr, dline_info, partitions, sector_power);
+    let balances = v.get_miner_balance(id_addr);
+    assert!(balances.initial_pledge > TokenAmount::zero());
+    let p_st = v.get_state::<PowerState>(*STORAGE_POWER_ACTOR_ADDR).unwrap();
+    assert_eq!(
+        StoragePower::from(seal_proof.sector_size().unwrap() as u64),
+        p_st.total_bytes_committed
+    );
+}
+
+fn create_miner(
+    v: &mut VM,
+    owner: Address,
+    worker: Address,
+    post_proof_type: RegisteredPoStProof,
+    balance: TokenAmount,
+) -> (Address, Address) {
+    let multiaddrs = vec![BytesDe("multiaddr".as_bytes().to_vec())];
+    let peer_id = "miner".as_bytes().to_vec();
+    let params = CreateMinerParams {
+        owner,
+        worker: worker,
+        window_post_proof_type: post_proof_type,
+        peer: peer_id.clone(),
+        multiaddrs: multiaddrs.clone(),
+    };
+
+    let res: CreateMinerReturn = v
+        .apply_message(
+            owner,
+            *STORAGE_POWER_ACTOR_ADDR,
+            balance,
+            PowerMethod::CreateMiner as u64,
+            params.clone(),
```

Comment:
it seems to be the last occurrence, shouldn't clippy complain that `clone` is not necessary?